src/mumble/pluginAPI/Plugin.h

```diff
@@ -0,0 +1,155 @@
+
+	// functions for init and de-init
+	
+	/// Gets called right after loading the plugin in order to let the plugin initialize.
+	///
+	/// @returns The status of the initialization. If everything went fine, return STATUS_OK
+	PLUGIN_EXPORT uint32_t init();
+	
+	/// Gets called when unloading the plugin in order to allow it to clean up after itself.
+	PLUGIN_EXPORT void shutdown();
+
+	/// Tells the plugin some basic information about the Mumble client loading it.
+	/// This function will be the first one that is being called on this plugin - even before it is decided whether to load
+	/// the plugin at all.
+	///
+	/// @param mumbleVersion The Version of the Mumble client
+	/// @param mumbleAPIVersion The Version of the plugin-API the Mumble client runs with
+	/// @param minimalExpectedAPIVersion The minimal Version the Mumble clients expects this plugin to meet in order to load it
+	PLUGIN_EXPORT void setMumbleInfo(Version_t mumbleVersion, Version_t mumbleAPIVersion, Version_t minimalExpectedAPIVersion);
+
+	// functions for general plugin info
+	
+	/// Gets the name of the plugin.
+	///
+	/// @param[out] nameBuffer The buffer into which the name of the plugin shall be copied by this function
+	/// @param bufferSize The size of the buffer - The length of the name of the plugin (including the terminating zero-byte) must not exceed
+	/// 	this length
+	PLUGIN_EXPORT void getName(char *nameBuffer, size_t bufferSize);
+
+	/// Gets the plugin's version in a display-ready representation.
+	///
+	/// @param[out] versionBuffer The buffer into which the version string shall be copied by this function
+	/// @param bufferSize The size of the buffer - The length of the version (including terminating zero-byte) must not exceed this length
+	PLUGIN_EXPORT void getDisplayVersion(char *versionBuffer, size_t bufferSize);
+
+	/// Gets the Version of this plugin
+	///
+	/// @returns The plugin's version
+	PLUGIN_EXPORT Version_t getVersion();
+
+	/// Gets the Version of the plugin-API this plugin intends to use.
+	/// Mumble will decide whether this plugin is loadable or not based on the return value of this function.
+	///
+	/// @return The respective API Version
+	PLUGIN_EXPORT Version_t getApiVersion();
+
+	/// Gets the name of the plugin author(s)
+	///
+	/// @param[out] authorBuffer The buffer into which the author name shall be copied by this function
+	/// @param bufferSize The size of the buffer - The length of the author name (including the terminating zero-byte) must not exceed
+	/// 	this length.
+	PLUGIN_EXPORT void getAuthor(char *authorBuffer, size_t bufferSize);
+
+	/// Gets the description of the plugin
+	///
+	/// @param[out] descriptionBuffer The buffer into which the description shall be copied by this function
+	/// @param bufferSize The siue of the buffer - The length of the description (including the terminating zero-byte) must not exceed
+	/// 	this length.
+	PLUGIN_EXPORT void getDescription(char *descriptionBuffer, size_t bufferSize);
```

Comment:
How do you decide on the buffer size for this? If the description is displayed in the scrollable box then there is no real upper limit from mumble side.
I think some things make more sense to let the plugin just return a string which we then take care of copying.

Reply:
And leaving it to the plugin to make sure it actually returns a valid pointer (e.g. not a pointer to something that is destroyed after the function scope exits)?

Reply:
Either that (which is what teamspeak does) or we could also tell them to allocate the buffer using the memoryalloc api function, thus we can delete it whenever we think were done with it.